recommend/bpmf.py

```diff
@@ -60,57 +60,73 @@ def __init__(self, n_user, n_item, n_feature, beta=2.0, beta_user=2.0,
         self.mu_user = np.zeros((n_feature, 1), dtype='float64')
         self.mu_item = np.zeros((n_feature, 1), dtype='float64')
 
-        self.alpha_user = np.eye(n_feature, dtype='float64')
+        self.alpha_user = np.eye(n_feature, dtype='float64')  # These are probably the Lambda matrices
         self.alpha_item = np.eye(n_feature, dtype='float64')
 
-        self.user_features_ = 0.3 * self.rand_state.rand(n_user, n_feature)
+        self.user_features_ = 0.3 * self.rand_state.rand(n_user, n_feature)  # initializes the user features randomly. Why 0.3??
         self.item_features_ = 0.3 * self.rand_state.rand(n_item, n_feature)
 
         # data state
         self.mean_rating_ = None
         self.ratings_csr_ = None
         self.ratings_csc_ = None
 
-    def fit(self, ratings, n_iters=50):
-        """training models"""
+    def fit(self, ratings, test_ratings=None, n_iters=50):
```

Comment:
pass `test_ratings` in the training step doesn't make much sense in the api point of view.
I think it is better to keep the average user/item matrix and use it in prediction directly?